servicex/query_cache.py

```diff
@@ -69,25 +69,75 @@ def transformed_results(self, transform: TransformRequest,
         )
 
     def cache_transform(self, record: TransformedResults):
+        transforms = Query()
         with self.lock:
-            if not self.contains_hash(record.hash):
-                self.db.insert(json.loads(record.model_dump_json()))
+            self.db.upsert(json.loads(record.model_dump_json()), transforms.hash == record.hash)
 
     def update_record(self, record: TransformedResults):
         transforms = Query()
         with self.lock:
             self.db.update(json.loads(record.model_dump_json()), transforms.hash == record.hash)
 
     def contains_hash(self, hash: str) -> bool:
+        """
+        Check if the cache has completed records for a hash
+        """
         transforms = Query()
         with self.lock:
-            records = self.db.search(transforms.hash == hash)
+            records = self.db.search((transforms.hash == hash) & (transforms.files.exists()))
         return len(records) > 0
 
+    def get_transform_request_status(self, hash_value: str) -> Optional[str]:
+        """
+        Get the status of a transform request
+        """
+        transform = Query()
+        with self.lock:
+            records = self.db.search((transform.hash == hash_value)
+                                     & (transform.status.exists()))
```

Comment:
Mild preference to do this check *after* the query, i.e. look up the request first (because having the request missing should be an error!) and then check if the field is there.

It's also true that you basically never use this function except to check if the result is `SUBMITTED`. Perhaps better to make the function `get_transform_request_submitted` and return a bool?

Reply:
I have changed the function to a bool. However, the request missing shouldn't raise an error since the the request is still not submitted in the case and it should return False rather than a error.

Reply:
Right, I see. Somewhat unfortunate naming of the function I suppose. It would be good to document this contract (it returns False if the transformation is not in the cache at all) in the docstring, since you rely on this behavior.